src/main/java/ch/njol/skript/classes/data/DefaultChangers.java

```diff
@@ -202,36 +204,19 @@ public void change(final Inventory[] invis, final @Nullable Object[] delta, fina
 						//$FALL-THROUGH$
 					case ADD:
 						assert delta != null;
-						
-						if(delta instanceof ItemStack[]) { // Old behavior - legacy code (is it used? no idea)
-							ItemStack[] items = (ItemStack[]) delta;
-							if(items.length > 36) {
-								return;
-							}
-							for (final Object d : delta) {
-								if (d instanceof Inventory) {
-									for (final ItemStack i : (Inventory) d) {
-										if (i != null)
-											invi.addItem(i);
-									}
-								} else {
-									((ItemType) d).addTo(invi);
-								}
-							}
-						} else {
-							for (final Object d : delta) {
-								if (d instanceof ItemStack) {
-									new ItemType((ItemStack) d).addTo(invi); // Can't imagine why would be ItemStack, but just in case...
-								} else if (d instanceof ItemType) {
-									((ItemType) d).addTo(invi);
-								} else if (d instanceof Block) {
-									new ItemType((Block) d).addTo(invi);
-								} else {
-									Skript.error("Can't " + d.toString() + " to an inventory!");
-								}
+						for (final Object d : delta) {
+							if (d instanceof ItemStack) {
+								new ItemType((ItemStack) d).addTo(invi); // Can't imagine why would be ItemStack, but just in case...
+							} else if (d instanceof ItemType) {
+								((ItemType) d).addTo(invi);
+							} else if (d instanceof Block) {
+								new ItemType((Block) d).addTo(invi);
+							} else if (d instanceof Inventory) {
+								invi.setContents(((Inventory) d).getContents());
```

Comment:
Copied from the Inventory#setContents Javadocs:

> Completely replaces the inventory's contents. Removes all existing contents and replaces it with the ItemStacks given in the array.
**Parameters:**
items - A complete replacement for the contents; the length must be less than or equal to getSize().
**Throws:**
IllegalArgumentException - If the array has more items than the inventory.

This means that when the player sets the contents to an incompatible inventory, it will give an error. Not sure if this is error is handled internally already but if not, you'll have to compare sizes.

Reply:
This was marked as resolved, but no changes were made?